server.js

```diff
@@ -7967,6 +7967,58 @@ cache(function (data, match, sendBadge, request) {
   });
 }));
 
+// Launchpad integration
+camp.route(/^\/launchpad\/ppa\/version\/([^/]+)\/([^/]+)\/([^/]+)\.(svg|png|gif|jpg|json)$/,
+cache(function(data, match, sendBadge, request) {
+  // The owner of the ppa
+  var user = match[1];
+  // The name of the ppa
+  var ppa = match[2];
+  // The package in the PPA to use
+  var name = match[3];
+  var format = match[4];
+
+  var badgeData = getBadgeData('launchpad', 'invalid');
+
+  // Make a request for packages in the PPA, returning only the published sources that match the the package
+  // Keep in mind that this will also return packages _starting_ with the package name
+  var apiUrl = 'https://api.launchpad.net/1.0/~' + user + '/+archive/' + ppa + '?ws.op=getPublishedSources&status=Published&source_name=' + name;
+  request(apiUrl, function(err, res, buffer) {
+    checkErrorResponse(badgeData, err, res, "not found");
+
+    try {
+      var data = JSON.parse(buffer);
+      var foundEntry = false;
+
+      // Loop through the results until we find the requested package
+      for (let entry of data.entries) {
+        if (entry.source_package_name == name) {
+          foundEntry = entry;
+          break;
+        }
+      }
+
+      // Error out if we can't find it
+      if (foundEntry === false) {
+        badgeData.text[1] = 'no package';
+        badgeData.colorscheme = 'red';
+        sendBadge(format, badgeData);
+        return;
+      }
+
+
+      badgeData.text[1] = foundEntry.source_package_version;
+      badgeData.colorscheme = 'blue';
```

Comment:
Could you could use [`versionText()`](https://github.com/badges/shields/blob/master/server.js#L706) and [`versionColor()`](https://github.com/badges/shields/blob/master/server.js#L707) here.

https://github.com/badges/shields/blob/master/server.js#L706-L707